e2e bigquerymultitable scenarios

itsmekumari opened the pull request:
e2e bigquerymultitable scenarios

priyabhatnagar25 committed:
BQMT e2e Tests Scenarios

Vipinofficial11 commented on src/e2e-test/features/bigquerymultitablesink/BiqQueryMultiTableSinkError.feature:
```diff
@@ -0,0 +1,77 @@
+# License for the specific language governing permissions and limitations under
+# the License.
+
+@BigQueryMultiTable_Sink
```
We can just make the tag as BQMT_SINK.

itsmekumari replied:
Change done.

Vipinofficial11 commented on src/e2e-test/features/bigquerymultitablesink/MultipleDatabaseTableToBigQueryMultiTableSink_WithMacro.feature:
```diff
@@ -0,0 +1,129 @@
+Feature: BigQueryMultiTable sink -Verification of BigQuery to BigQueryMultiTable successful data transfer using macros
+
+  @MULTIPLEDATABASETABLE_SOURCE_TEST
+  Scenario:Verify data is getting transferred from BigQuery to BQMT sink with all datatypes using macros
```
In the scenario it says we are running a pipeline from BQ to BQMT but we are selecting source as ```Multiple Database Tables``` which one is it ?

itsmekumari replied:
Changed to Multiple Database Tables

Vipinofficial11 commented on src/e2e-test/features/bigquerymultitablesink/MultipleDatabaseTableToBigQueryMultiTableSink_WithMacro.feature:
```diff
@@ -0,0 +1,129 @@
+
+  @MULTIPLEDATABASETABLE_SOURCE_TEST
+  Scenario:Verify data is getting transferred from BigQuery to BQMT sink with split field using macros
+    Given Open Datafusion Project to configure pipeline
```
Same comment here related to BigQuery

itsmekumari replied:
Changed to Multiple Database Tables

Vipinofficial11 commented on src/e2e-test/features/bigquerymultitablesink/MultipleDatabaseTableToBiqQueryMultiTableSink.feature:
```diff
@@ -0,0 +1,102 @@
+    Then Validate the values of records transferred to BQMT sink is equal to the value from source MultiDatabase table
+
+  @MULTIPLEDATABASETABLE_SOURCE_DATATYPES_TEST
+  Scenario:Verify data is getting transferred from BigQuery to BQMT sink with split field
```
BigQuery -> ```Multiple Database Tables```

itsmekumari replied:
Done.

Vipinofficial11 commented on src/e2e-test/java/io/cdap/plugin/bigquerymultitable/locators/BigQueryMultiTableLocators.java:
```diff
@@ -0,0 +1,36 @@
+  public static WebElement datasetProject;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-cy='location']")
+  public static WebElement location;
```
Where are we using these 2 locators I don't see them in actions.

itsmekumari replied:
removed

Vipinofficial11 commented on src/e2e-test/java/io/cdap/plugin/bigquerymultitable/locators/BigQueryMultiTableLocators.java:
```diff
@@ -0,0 +1,36 @@
+  @FindBy(how = How.XPATH, using = "//input[@data-cy='location']")
+  public static WebElement location;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy='sqlStatements']//*[@data-cy='key']/input")
```
If this is a new property can we make a common locator out of this and add this in the framework ?

itsmekumari replied:
@priyabhatnagar25 A generic step is added similar type of locator for BQ framework PR, I believe we can use it.

Vipinofficial11 commented on src/e2e-test/java/io/cdap/plugin/bigquerymultitable/stepdesign/BQMultiTableValidation.java:
```diff
@@ -0,0 +1,231 @@
+   * @throws ClassNotFoundException If the class specified is not found.
+   */
+
+  public static boolean validateMySqlToBQRecordValues(String sourceTable) throws IOException, InterruptedException,
```
So we are using Multiple Database table plugin here but we are using a SQL Database/Table. It would be good if we can add few comments and context here describing this in this class.

Vipinofficial11 commented on src/e2e-test/java/io/cdap/plugin/bigquerymultitable/stepdesign/BQMultiTableValidation.java:
```diff
@@ -0,0 +1,231 @@
+   * @throws InterruptedException If the current thread is interrupted while waiting.
+   */
+
+  public static List<String> getTableByName() throws IOException, InterruptedException {
```
change the method name to getTableNameFromMySQL()

itsmekumari replied:
change done.

Vipinofficial11 commented on src/e2e-test/java/io/cdap/plugin/bigquerymultitable/stepdesign/BQMultiTableValidation.java:
```diff
@@ -0,0 +1,231 @@
+  public static List<String> getTableByName() throws IOException, InterruptedException {
+    List<String> tableNames = new ArrayList<>();
+    List<String> targetTableNames = Arrays.asList(tabA, tabB);
+    TableResult tableResult = getTableNamesFromDataSet();
```
Name it as getTableNamesFromBQDataSet()

itsmekumari replied:
Change done

bharatgulati commented on src/e2e-test/java/io/cdap/plugin/bigquerymultitable/stepdesign/BQMultiTableValidation.java:
```diff
@@ -0,0 +1,231 @@
+          case Types.VARCHAR:
+            String sourceString = rsSource.getString(currentColumnCount);
+//                        String sour = bigQueryData.get(jsonObjectIdx).get(columnTypeName).getAsString();
+//                        String sou2 = bigQueryData.get(jsonObjectIdx).get(columnType)
```
Remove commented line no. 155 & 156.

poojantcs commented on src/e2e-test/java/io/cdap/plugin/bigquerymultitable/actions/BigQueryMultiTableActions.java:
```diff
@@ -0,0 +1,35 @@
+    SeleniumHelper.getPropertiesLocators(BigQueryMultiTableLocators.class);
+  }
+
+    public static void enterSqlStatement(String value) {
```
Move the locator and the corresponding action to the framework repo, if it's not already there.

poojantcs commented on src/e2e-test/java/io/cdap/plugin/bigquerymultitable/stepdesign/MultiTableClient.java:
```diff
@@ -0,0 +1,49 @@
+  public static void createSourceDatatypesTable(String sourceTable) throws SQLException, ClassNotFoundException {
+    try (Connection connect = getMysqlConnection(); Statement statement = connect.createStatement()) {
+      String createTableQuery = "CREATE TABLE " + sourceTable +
+        "(EmployeeID int, LastName varchar(255), City varchar(255), tablename varchar(255))";
```
Can we make this parameterized?

priyabhatnagar25 committed:
Removed Issue tag Plugin-1669 in Runtime scenarios and test runner file,
updated correct conn string,
removed space in yml,
removed verify goal in pom.
changed sql connector java version to run in github actions and
bqmt secret updated

Vipinofficial11 committed:
Debug runner path failures for BQMT
